File: csharp/triangle/Triangle.cs
using System;

public static class Triangle
{
    public static bool IsScalene(double side1, double side2, double side3){
        if (!isTriangle(side1, side2, side3)){
            return false;
        }
        if (side1 != side2 && side2 != side3 && side1 != side3){
            return true;
        }
        return false;

    }

    public static bool IsIsosceles(double side1, double side2, double side3) 
    {
        if (!isTriangle(side1, side2, side3)){
            return false;
        }
        if (side1 == side2 || side2 == side3 || side1 == side3){
            return true;
        }
        return false;
    }

    public static bool IsEquilateral(double side1, double side2, double side3) {
        if (!isTriangle(side1, side2, side3)){
            return false;
        }
        if (side1 == side2 && side2 == side3){
            return true;
        }
        return false;

    }

// Determine whether the object even is a triangle
    public static bool isTriangle(double side1, double side2, double side3) =>  side1 > 0 && 
                                                                                side2 > 0 && 
                                                                                side2 > 0 && 
                                                                                side1 + side2 >= side3 &&
                                                                                side2 + side3 >= side1 && 
                                                                                side1 + side3 >= side2;

    }

File: c/triangle/triangle.c
#include "triangle.h"

bool is_triangle(triangle_t t) {
  if (t.a == 0 || t.b == 0 || t.c == 0)
    return false;

  if (t.a + t.b < t.c || t.a + t.c < t.b || t.b + t.c < t.a)
    return false;

  return true;
}
bool is_equilateral(triangle_t t) {
  return is_triangle(t) && (t.a == t.b && t.b == t.c);
}

bool is_isosceles(triangle_t t) {
  return is_triangle(t) && (t.a == t.b || t.a == t.c || t.b == t.c);
}
bool is_scalene(triangle_t t) {
  return is_triangle(t) && (t.a != t.b && t.a != t.c && t.b != t.c);
}
